extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputFormat.java

```diff
@@ -73,41 +74,22 @@ public boolean isSplittable()
   @Override
   public InputEntityReader createReader(InputRowSchema inputRowSchema, InputEntity source, File temporaryDirectory)
   {
-    // assume InputEntity is always defined in a single classloader (the kafka-indexing-service classloader)
-    // so we only have to look it up once. To be completely correct we should cache the method based on classloader
-    if (getHeaderMethod == null) {
-      getHeaderMethod = KafkaUtils.lookupGetHeaderMethod(
-          source.getClass().getClassLoader(),
-          OpenCensusProtobufInputFormat.VERSION_HEADER_KEY
-      );
+    // Sampler passes a KafkaRecordEntity directly, while the normal code path wraps the same entity in a
+    // SettableByteEntity
```

Comment:
it might be worth making a change upstream to have the sampler also pass in a SettableByteEntity to avoid having to handle this edge.